aioshelly/common.py

```diff
@@ -137,3 +139,23 @@ def shelly_supported_firmware(result: dict[str, Any]) -> bool:
     # We compare firmware release dates because Shelly version numbering is
     # inconsistent, sometimes the word is used as the version number.
     return int(match[0]) >= fw_ver
+
+
+async def trigger_ota_http(session: aiohttp.ClientSession, host: str, gen: int) -> bool:
+    """Trigger a firmware update via OTA http endpoint."""
+    if gen in BLOCK_GENERATIONS:
+        path = "ota"
+    else:
+        path = "rpc/Shelly.Update"
+    try:
+        await session.get(
```

Comment:
What about authorization if device is password protected?

Reply:
Am I wrong that OTA doesn't need auth ?

Reply:
OTA needs auth same as other functions